### PR TITLE
Port managed Array.Copy/Clear from CoreCLR version

### PR DESCRIPTION
**Pull request by jkotas**
Also fixes a few infrastructure issues that I have run into along the way.

### CODE REVIEW

**Review by MichalStrehovsky — APPROVED** [review]
Thanks!

File: src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs [hunk 1]
```diff
@@ -227,30 +227,74 @@ public void Initialize()
...
         {
-            if (!RuntimeImports.TryArrayCopy(sourceArray, 0, destinationArray, 0, length))
-                CopyImpl(sourceArray, 0, destinationArray, 0, length, false);
+            if (sourceArray is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.sourceArray);
+            if (destinationArray is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.destinationArray);
+
+            EEType* pEEType = sourceArray.EEType;
+            if (pEEType == destinationArray.EEType &&
```
[the remark this answers is not in the document]
Comment (jkotas) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs]: I kelp the type equivalency check (the faster version). Switching to EETypePtr adds a bit of extra overhead. RyuJIT is not able to optimize out the wrapper completely. We can deal with that problem separately accross the board (by either fixing RyuJIT or eliminating the EETypePtr wrapper).

[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs]: > RyuJIT is not able to optimize out the wrapper completely

Ah, that's unfortunate.

[the remark this answers is not in the document]
Comment (jkotas) [reply src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs]: Opened https://github.com/dotnet/runtimelab/issues/232

**Pull request merged**